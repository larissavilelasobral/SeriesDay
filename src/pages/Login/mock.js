import { googleLogin, signIn } from '../../services/index.js';

export const Login = () => {
  const login = document.createElement('div');
  login.innerHTML = `
    <input class="input" id="email" type="email" autocomplete="on" placeholder="E-mail" required>
    <input class="input" id="password" type="password" autocomplete="on" placeholder="Senha" required>
    <button id="signin-button" class="signin-button buttons">Entrar</button>

    <button id="google-button" class="google-button buttons">
        <img class="google-icon-btn" src="../../assets/google-icon.svg" alt="Ícone do Google"/>
            Entrar com Conta Google
    </button>
    `;

  // VARIAVEIS
  const email = login.querySelector('#email');
  const password = login.querySelector('#password');
  const loginError = login.querySelector('#loginError');
  const signInButton = login.querySelector('#signin-button');

  // LOGIN DE USUARIOS EXISTENTES POR EMAIL E SENHA
  signInButton.addEventListener('click', (e) => {
    e.preventDefault();
    if (email.value) {
      signIn(email.value, password.value)
        .then(() => {
          window.location.hash = 'timeline'; // ir para o feed
        })
        .catch((error) => {
          const errorCode = error.code;
          if (errorCode === 'auth/invalid-email') {
            loginError.style.color = 'red"';
            loginError.innerHTML = 'Não há registro de usuário correspondente a este e-mail';
          } else if (errorCode === 'auth/wrong-password') {
            loginError.style.color = 'red';
            loginError.innerHTML = 'Senha inválida';
          }
        });
    }
  });

  //  AUTENTICAÇÃO COM CONTA GOOGLE
  const googleButton = login.querySelector('#google-button');

  googleButton.addEventListener('click', (e) => {
    e.preventDefault();
    googleLogin();
  });

  return login;
};
