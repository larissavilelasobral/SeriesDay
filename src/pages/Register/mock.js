import { saveUserUpdate, saveUser, registerUser } from '../../services/index.js';

export const Register = () => {
  const register = document.createElement('div');
  register.innerHTML = `
  <link rel="stylesheet" href="./pages/Register/style.css" />

  <main class="box">
      <div class="container">
        <div class="banner">
            <img src="assets/logo.png" alt="Logo">
            <div class="title-container">
                <h1 class="title">SeriesDay</h1>
                <h3 class="subtitle">review de s√©ries</h3>
            </div>
        </div>
          
        <section>
            <form class="form">
                <input class="input" id="name" type="name" autocomplete="on" placeholder="üóí  Nome Completo" required>
                <input class="input" id="email" type="email autocomplete="on" placeholder="‚úâ  E-mail" required>
                <p id="email-error" class="error-message font-work"></p>

                <input class="input" id="password" type="password" autocomplete="on" placeholder="‚öô  Senha" required>

                <p class="error-message font-work" id="password-length"></p>

                <input class="input" id="password-confirm" type="password" autocomplete="on" placeholder="‚öô  Confirmar Senha" required>

                <p class="error-message font-work" id="password-error"></p>
            </form>
            <button id="signup-button-register" class="buttons register-button">Cadastrar-se</button>
            <button id="gobackButton" class="goback-button">
                <img src="./assets/arrow.png" alt="√çcone de Seta"> 
            </button>
        </section>
      </div>
  </main>
  
    `;

  const profileName = register.querySelector('#name');
  const email = register.querySelector('#email');
  const password = register.querySelector('#password');
  const emailError = register.querySelector('#email-error');

  const signUpButtonRegister = register.querySelector('#signup-button-register');

  // INPUTS PARA CADASTRO
  signUpButtonRegister.addEventListener('click', (e) => {
    e.preventDefault();
    registerUser(email.value, password.value)
      .then((userUpdate) => {
        saveUserUpdate(profileName.value);
        saveUser(userUpdate.user, email.value, profileName.value);
      })
      .catch((error) => {
        const errorCode = error.code;
        if (errorCode === 'auth/email-already-in-use') {
          emailError.style.color = 'red';
          emailError.innerHTML = 'E-mail j√° cadastrado';
        } else if (errorCode === 'auth/invalid-email') {
          emailError.style.color = 'red';
          emailError.innerHTML = 'Insira um e-mail v√°lido';
        }
      });
  });

  return register;
};
