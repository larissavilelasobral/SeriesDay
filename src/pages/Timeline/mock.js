import { signOut } from '../../services/index.js';

export const Timeline = () => {
  const timeline = document.createElement('div');
  timeline.innerHTML = `
    <button id="signout-button" class="signout-button buttons">
      <img src="./assets/exit-black.png" alt="Ícone de Saída">
    </button>

    <button id="desktop-signout-button" class="signout-button desktop-signout-button buttons">
      <img src="./assets/exit-white.png" alt="Ícone de Saída">
    </button>

    <div id="deleteBtn">
      <button class="deletePost-btn timeline-buttons font-work visibility-hidden">
        <img src="./assets/trash.png" alt="Ícone de Lixeira">
      </button>
    </div>
  `;

  // Sair da conta do usuário (MOBILE)
  timeline.querySelector('#signout-button').addEventListener('click', (e) => {
    e.preventDefault();
    signOut();
  });

  // Sair da conta do usuário (DESKTOP)
  timeline.querySelector('.desktop-signout-button').addEventListener('click', (e) => {
    e.preventDefault();
    signOut();
  });

  return timeline;
};
