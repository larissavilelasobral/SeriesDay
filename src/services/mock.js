// Configurando as autenticações
function googleLogin() {
  const provider = new firebase.auth.GoogleAuthProvider();
  firebase.auth().signInWithPopup(provider)
    .then((result) => {
      const user = result.user;
      firebase.firestore().collection('users').doc(user.email)
        .set({
          name: user.displayName,
          id: user.uid,
          photo: user.photoURL,
        }, { merge: true });
    })
    .catch((error) => {
      const errorCode = error.code;
      if (errorCode === 'auth/account-exists-with-different-credential') {
        // eslint-disable-next-line no-alert
        alert('Essa conta já existe com uma credencial diferente');
      }
    });
}
googleLogin();

function signOut() {
  firebase.auth().signOut();
  window.location.hash = '';
  window.location.reload();
}
signOut();

function saveUserUpdate(name) {
  firebase.auth().currentUser.updateProfile({
    displayName: name,
  })
    .then(() => true)
    .catch((error) => error);
}
saveUserUpdate();

function saveUser(user, userEmail, userName) {
  firebase.firestore().collection('users').doc(userEmail).set({
    id: user.uid,
    name: userName,
    email: userEmail,
    photo: '',
  })
    .then(() => true)
    .catch((error) => error);
}
saveUser();

function registerUser(email, password) {
  firebase.auth().createUserWithEmailAndPassword(email, password);
}
registerUser();

function signIn(email, password) {
  firebase.auth().signInWithEmailAndPassword(email, password);
}
signIn();

function deletePost(id, postsCollection) {
  postsCollection
    .doc(id)
    .delete();
}
deletePost();

function editPost(newPost, id, postsCollection) {
  postsCollection
    .doc(id)
    .update({
      text: newPost,
    });
}
editPost();
