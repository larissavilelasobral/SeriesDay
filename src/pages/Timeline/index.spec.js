import * as services from '../../services/index.js';
import { Timeline } from './mock.js';

services.signOut = jest.fn(() => Promise.resolve(true));
const timelinePage = Timeline();

describe('Timeline', () => {
  it('should be a function', () => {
    expect(typeof Timeline).toBe('function');
  });

  it('should load the page', () => {
    expect(Timeline()).toMatchSnapshot();
  });
});

describe('signOut', () => {
  it('should be a function', () => {
    expect(typeof services.signOut).toBe('function');
  });

  it('When the user clicks the button, he should sign out from the site on mobile version', () => {
    timelinePage.querySelector('#signout-button').dispatchEvent(new Event('click'));
    expect(services.signOut).toBeCalled();
  });

  it('When the user clicks the button, he should sign out from the site on desktop version ', () => {
    timelinePage.querySelector('#desktop-signout-button').dispatchEvent(new Event('click'));
    expect(services.signOut).toBeCalled();
  });
});

describe('deletePost', () => {
  it('should be a function', () => {
    expect(typeof services.deletePost).toBe('function');
  });
});

describe('editPost', () => {
  it('should be a function', () => {
    expect(typeof services.editPost).toBe('function');
  });
});
