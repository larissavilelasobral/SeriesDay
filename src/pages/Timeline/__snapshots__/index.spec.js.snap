// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline should load the page 1`] = `
<div>
  
    
  <button
    class="signout-button buttons"
    id="signout-button"
  >
    
      
    <img
      alt="Ícone de Saída"
      src="./assets/exit-black.png"
    />
    
    
  </button>
  

    
  <button
    class="signout-button desktop-signout-button buttons"
    id="desktop-signout-button"
  >
    
      
    <img
      alt="Ícone de Saída"
      src="./assets/exit-white.png"
    />
    
    
  </button>
  

    
  <div
    id="deleteBtn"
  >
    
      
    <button
      class="deletePost-btn timeline-buttons font-work visibility-hidden"
    >
      
        
      <img
        alt="Ícone de Lixeira"
        src="./assets/trash.png"
      />
      
      
    </button>
    
    
  </div>
  
  
</div>
`;
